Framework/PythonInterface/mantid/plots/datafunctions.py

```diff
@@ -391,25 +391,27 @@ def get_bin_indices(workspace):
         return indices
 
 
-def get_bins(workspace, wkspIndex, withDy=False):
+def get_bins(workspace, bin_index, withDy=False):
     """
     Extract all the bins for a spectrum
 
     :param workspace: a Workspace2D or an EventWorkspace
-    :param wkspIndex: workspace index
+    :param bin_index: the index of a bin
     :param withDy: if True, it will return the error in the "counts", otherwise None
 
     """
-    bins = get_bin_indices(workspace)
-    y_values = []
+    indices = get_bin_indices(workspace)
+    x_values, y_values = [], []
     dy = [] if withDy else None
-    for bin_index in bins:
-        y_values.append(workspace.readY(int(bin_index))[wkspIndex])
-        if withDy:
-            dy.append(workspace.readE(int(bin_index))[wkspIndex])
-
+    for row_index in indices:
```

Comment:
I have tried plotting a bin in some SANS data which contained 50k spectra, and it took about 5 seconds to plot the bin. Do you think this needs to be sped up? Any ideas how this might be done?

Reply:
That is a little slow. One option would be to add some C++ code to be able to retrieve the "column" for a given index. I think that is heading outside of the scope of this PR though.

Reply:
As long as it took that amount of time before these changes (and is it a Release build)?

Reply:
It takes the same amount of time as before. And on Release it takes just over a 1 second

Reply:
Ok. I'm not sure that's worth the effort of trying to speed up at present.